### PR TITLE
Feature/domain events

### PR DESCRIPTION
**Pull request by vkirichenko**

### CODE REVIEW

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/RetryService.java [hunk 1]
```diff
@@ -60,11 +61,14 @@ public class RetryService {
 
     @PostConstruct
     public void postConstruct() {
-        tenantListRepository.getTenants().forEach(tenant -> {
-            tenantUtils.doInTenantContext(() -> {
-                self.rescheduleAllEvents();
-            }, tenant);
-        });
+        //using new thread so tenantUtils.doInTenantContext() would not destroy current tenant context in main thread
+        new Thread(() ->
```
Comment (vkirichenko) [inline src/main/java/com/icthh/xm/tmf/ms/activation/service/RetryService.java:65]: @valeriibilenko we need to wait until this thread is completed

File: src/main/java/com/icthh/xm/tmf/ms/activation/utils/TenantUtils.java [hunk 2]
```diff
@@ -25,6 +28,17 @@ public String getTenantKey() {
         return TenantContextUtils.getRequiredTenantKeyValue(tenantContextHolder);
     }
 
+    /**
+     * Run code in context with specified tenant. Keep in mind, that after task is executed tenant context
+     * will be destroyed.
+     * NOTICE: If run before application start up: consider to run in new thread (or setting
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/tmf/ms/activation/utils/TenantUtils.java:34]: as we skipped the idea bout new thread then need to correct this readme

Comment (valeriibilenko) [reply src/main/java/com/icthh/xm/tmf/ms/activation/utils/TenantUtils.java]: @amedvedchuk done

**Review by amedvedchuk — APPROVED** [review]

File: src/main/java/com/icthh/xm/tmf/ms/activation/config/LepContext.java [hunk 3]
```diff
@@ -31,11 +32,14 @@ public class LepContext {
...
 
     public ApplicationContext applicationContext;
+    private OutboxTransportService outboxTransportService;
 
     public static class LepServices {
         public TenantConfigService tenantConfigService;
         public MailService mailService;
         public SagaService sagaService;
+        private EventPublisher eventPublisher;
+        private DomainEventFactory domainEventFactory;
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/activation/config/LepContext.java:42]: why private?

File: src/main/java/com/icthh/xm/tmf/ms/activation/config/LepContext.java [hunk 4]
```diff
@@ -31,11 +32,14 @@ public class LepContext {
     public LepTemplates templates;
 
     public ApplicationContext applicationContext;
+    private OutboxTransportService outboxTransportService;
```
[the remark this answers is not in the document]
Comment (vkirichenko) [reply src/main/java/com/icthh/xm/tmf/ms/activation/config/LepContext.java]: @sergeysenja1992 fixed

**Review by sergeysenja1992 — APPROVED** [review]

**Pull request merged**